collection.go

```diff
@@ -433,3 +574,45 @@ func (c *Collection) getDocPath(docID string) string {
 	}
 	return docPath
 }
+
+func reorderResults(results []Result, negativeVector []float32, negativeReorderStrength float32) ([]Result, error) {
+	if len(results) == 0 {
+		return results, nil
+	}
+
+	// Calculate cosine similarity between negative vector and each result
+	for i := range results {
+		sim, err := dotProduct(negativeVector, results[i].Embedding)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't calculate dot product: %w", err)
+		}
+		results[i].Similarity -= sim * negativeReorderStrength
```

Comment:
Another question: This value is then based *only* on the negative similarity, not taking the original into account, right? Is that intended? In the PR description you wrote:

> Then re-order the result by multiplying the similarity by the dot product with the negative vector and some constant.

Where it sounded like the original similarity value is part of the multiplication.

As a follow-up question to this: If it's just the negative similarity, is the multiplication with the constant then required at all?